src/iotjs_module_buffer.cpp

```diff
@@ -216,6 +235,10 @@ JHANDLER_FUNCTION(Write) {
 
   BufferWrap* buffer_wrap = BufferWrap::FromJBufferBuiltin(*jbuiltin);
 
+  offset = boundRange(offset, 0, buffer_wrap->length());
```

Comment:
If `write` function is called from buffer.js, out-of-range offset will throw RangeError, so it's OK because this line cannot be called.

And if `write` function is called directly from _builtin.write, I think it is more reasonable not to write any bytes in the buffer, if input offset was out-of-range. 
If we use `buffer_wrap->length() - 1`, `buffer_wrap->Copy()` will write 1 byte into buffer even if the offset was invalid.